src/sensor/parser-ping.cpp

```diff
@@ -0,0 +1,20 @@
+#include "parser-ping.h"
+
+void PingParserExt::parseBuffer(const QByteArray& data)
+{
+    for(int i = 0; i < data.length(); i++) {
+        PingParser::ParseState state = parser_.parseByte(data.at(i));
+        if (state == PingParser::ParseState::NEW_MESSAGE) {
+            parsed++;
+            emit newMessage(ping_message(parser_.rxMessage));
+        } else if (state == PingParser::ParseState::ERROR) {
+            errors++;
+            emit parseError();
+        }
+    }
+}
```

Comment:
This function does not return any type of information related to the correct parser of a message, it should return a bool or a state condition to allow synchronous logic.

Reply:
This function is explicitly for asynchronous use only: https://github.com/bluerobotics/ping-viewer/blob/050fe06c7bba575b5a824c9406eba6c1aca2a476/src/sensor/parser.h#L17

Synchronous operation is only allowed with the parseByte() function  (as in arduino implementation).

Reply:
I've updated this class to allow synchronous operation according to your recommendation, but this function remains void return.

Reply:
Because `parseBuffer` might parse more than one message in a single call. It was created to add async signalling to the class.

Reply:
We could make it bool return, but I don't see any use case to apply it to.